server/src/QRServer/db/connector.py

```diff
@@ -190,6 +190,14 @@ async def add_match_result(self, match_result: DbMatchReport):
                 match_result.is_void
             ))
 
+        ranked_only = self.config.leaderboards_ranked_only.get()
+        include_void = self.config.leaderboards_include_void.get()
+
+        if (match_result.is_ranked or not ranked_only) and (not match_result.is_void or include_void):
+            await self.update_users_rating(
```

Comment:
this will call `commit()` within, which most likely will mess with the `commit` in this function, possibly we need to set up multiple separate transactions, one on this functions' level and the other one within (or remove commit from this function, but it was there to avoid collisions (with `rollback` mechanism)

Reply:
TODO

Reply:
Update: `update_users_rating` does not call `commit()`